sfge/src/main/java/com/salesforce/rules/fls/apex/AbstractFlsVisitor.java

```diff
@@ -173,4 +147,70 @@ private void recursivelyProcessConditionChildren(
     }
+
+    /**
+     * Indicates whether a boolean expression should be processed recursively, based on the nature
+     * of its operator, its children, and whether its satisfaction is expected for this path.
+     */
+    // TODO: This implementation is somewhat simplistic, and there are definitely
+    //       cases that it doesn't support. These cases will lead to false positives.
+    //       If unsupported edge cases crop up frequently in the wild, we should
+    //       enhance this to be more intelligent.
+    private boolean booleanExprRequiresRecursion(
+            BooleanExpressionVertex booleanExpressionVertex, boolean expectingSatisfied) {
+        BaseSFVertex lhs = booleanExpressionVertex.getLhs();
+        BaseSFVertex rhs = booleanExpressionVertex.getRhs();
+        if (booleanExpressionVertex.isOperatorAnd() && expectingSatisfied) {
+            // An AND expression expecting satisfaction should always be
+            // recursively processed, since it's only true if both sides are true.
+            return true;
+        } else if (booleanExpressionVertex.isOperatorOr()) {
+            // An OR expression should be recursively processed if it's not supposed
+            // to be satisfied (since that's only true if neither side is true),
+            // or if either side is literally false, since it requires the other
+            // side to be true.
+            return !expectingSatisfied || isLiterallyFalse(lhs) || isLiterallyFalse(rhs);
+        } else if (booleanExpressionVertex.isOperatorEquals()) {
+            // An EQUALS expression should be recursively processed if it's supposed
+            // to be satisfied and either side is literally true, or if it's not
+            // supposed to be satisfied and either side is literally false.
+            return (expectingSatisfied && (isLiterallyTrue(lhs) || isLiterallyTrue(rhs)))
+                    || (!expectingSatisfied && (isLiterallyFalse(lhs) || isLiterallyFalse(rhs)));
+        } else if (booleanExpressionVertex.isOperatorNotEquals()) {
+            // A NOT-EQUALS expression should be recursively processed if it's supposed
+            // to be satisfied and either side is false true, or if it's not
+            // supposed to be satisfied and either side is literally true.
+            return (!expectingSatisfied && (isLiterallyTrue(lhs) || isLiterallyTrue(rhs)))
+                    || (expectingSatisfied && (isLiterallyFalse(lhs) || isLiterallyFalse(rhs)));
+        } else {
+            // We can't say anything else for certain.
+            return false;
+        }
+    }
+
+    private boolean isLiterallyTrue(BaseSFVertex vertex) {
+        // We can drill down into a negation.
+        if (vertex instanceof PrefixExpressionVertex) {
+            PrefixExpressionVertex prefix = (PrefixExpressionVertex) vertex;
+            if (prefix.isOperatorNegation()) {
+                // If the inner vertex is literally false, its negation is literally true.
```

Comment:
In addition to your helpful text description, would be useful to have example code segments to visualize the process more easily.

Reply:
Not just here, but also in the if/else ladder above.